executor/aggregate.go

```diff
@@ -827,33 +830,54 @@ func (e *StreamAggExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	return nil
 }
 
-func (e *StreamAggExec) consumeOneGroup(ctx context.Context, chk *chunk.Chunk) error {
-	for !e.executed {
-		if err := e.fetchChildIfNecessary(ctx, chk); err != nil {
+func (e *StreamAggExec) consumeOneGroup(ctx context.Context, chk *chunk.Chunk) (err error) {
+	if e.groupChecker.isExhausted() {
+		if err = e.fetchChild(ctx, chk); err != nil {
 			return err
 		}
-		for ; e.inputRow != e.inputIter.End(); e.inputRow = e.inputIter.Next() {
-			meetNewGroup, err := e.groupChecker.meetNewGroup(e.inputRow)
+		if !e.executed && !chk.IsFull() {
+			_, err := e.groupChecker.splitIntoGroups(e.childResult)
 			if err != nil {
 				return err
 			}
-			if meetNewGroup {
-				err := e.consumeGroupRows()
-				if err != nil {
-					return err
-				}
-				err = e.appendResult2Chunk(chk)
-				if err != nil {
-					return err
-				}
-			}
-			e.groupRows = append(e.groupRows, e.inputRow)
-			if meetNewGroup {
-				e.inputRow = e.inputIter.Next()
-				return nil
+		} else {
+			return nil
+		}
+	}
+	begin, end := e.groupChecker.getNextGroup()
+	e.groupRows = e.groupRows[:0]
+	for i := begin; i < end; i++ {
+		e.groupRows = append(e.groupRows, e.childResult.GetRow(i))
+	}
+
+	for end == e.childResult.NumRows() {
+		end = -1
+		if err = e.fetchChild(ctx, chk); err != nil || e.executed {
+			return err
+		}
+
+		areSameGroup, err := e.groupChecker.splitIntoGroups(e.childResult)
+		if err != nil {
+			return err
+		}
+
+		if areSameGroup {
+			begin, end = e.groupChecker.getNextGroup()
```

Comment:
what if `end == e.childResult.NumRows()` here?

Reply:
When the ```end == e.childResult.NumRows()```, it means the new chunk only has one group which is the same group as the previous group. It will consume the group and get the next chunk(line 853).